src/celeritas/phys/PhysicsStepUtils.hh

```diff
@@ -269,28 +269,40 @@ select_discrete_interaction(MaterialView const& material,
     CELER_EXPECT(physics.interaction_mfp() <= 0);
     CELER_EXPECT(pstep.macro_xs() > 0);
 
-    // Sample ParticleProcessId from physics.per_process_xs()
-    ParticleProcessId ppid = celeritas::make_selector(
-        [&pstep](ParticleProcessId ppid) { return pstep.per_process_xs(ppid); },
-        ParticleProcessId{physics.num_particle_processes()},
-        pstep.macro_xs())(rng);
```

Comment:
Could you break `find_ppid` into a separate helper function, returning `{}` for rejection? Also I noticed that the integral rejection should probably use the rejection sampler for error checking + clarity.

Reply:
I think we weren't using the rejection sampler there because it's possible for the post-step xs to be larger than the estimate of the max xs over the step.

Reply:
No you're right, if the estimate of the maximum is wrong the sampling won't be correct. I think the only time this should happen though if the cross section has more than one peak and the second local maximum is within the pre- and post-step energy. We could mitigate this by storing multiple maximum xs values if there are multiple peaks (it looks like Geant4 now has a [`G4CrossSectionType`](https://github.com/Geant4/geant4/blob/32390e802bd84a8b85ac02a90a3d31cf205dd5b7/source/processes/electromagnetic/utils/include/G4EmTableType.hh#L54) that let's you specify up to two peaks). My understanding though is that the cross sections for these energy loss processes tend to be unimodal.